File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/services/PaymentService.kt
package com.desktopkitchen.pos.services

import com.desktopkitchen.pos.models.CashPaymentRequest
import com.desktopkitchen.pos.models.ConfirmPaymentRequest
import com.desktopkitchen.pos.models.CreatePaymentIntentRequest
import com.desktopkitchen.pos.models.PaymentIntent
import com.desktopkitchen.pos.models.PaymentStatus
import com.desktopkitchen.pos.networking.api.PaymentApi
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PaymentService @Inject constructor(
    private val paymentApi: PaymentApi
) {
    suspend fun createIntent(orderId: Int, tip: Double? = null): PaymentIntent =
        paymentApi.createIntent(CreatePaymentIntentRequest(order_id = orderId, tip = tip))

    suspend fun confirm(orderId: Int, paymentIntentId: String) {
        paymentApi.confirm(ConfirmPaymentRequest(order_id = orderId, payment_intent_id = paymentIntentId))
    }

    suspend fun cashPayment(orderId: Int, tip: Double? = null) {
        paymentApi.cashPayment(CashPaymentRequest(order_id = orderId, tip = tip))
    }

    suspend fun getStatus(orderId: Int): PaymentStatus = paymentApi.getStatus(orderId)
}

File: apps/pos/native-windows/DesktopKitchenPOS/Services/PaymentService.cs
using DesktopKitchenPOS.Models;
using DesktopKitchenPOS.Networking.Endpoints;

namespace DesktopKitchenPOS.Services;

public class PaymentService
{
    public async Task<PaymentIntent> CreateIntentAsync(int orderId, double? tip = null)
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<PaymentIntent>(
            HttpMethod.Post, PaymentEndpoints.CreateIntent,
            new CreatePaymentIntentRequest { OrderId = orderId, Tip = tip });
    }

    public async Task ConfirmAsync(int orderId, string paymentIntentId)
    {
        await DesktopKitchenPOS.App.Api.RequestVoidAsync(
            HttpMethod.Post, PaymentEndpoints.Confirm,
            new ConfirmPaymentRequest { OrderId = orderId, PaymentIntentId = paymentIntentId });
    }

    public async Task CashPaymentAsync(int orderId, double? tip = null)
    {
        await DesktopKitchenPOS.App.Api.RequestVoidAsync(
            HttpMethod.Post, PaymentEndpoints.Cash,
            new CashPaymentRequest { OrderId = orderId, Tip = tip });
    }

    public async Task<PaymentStatus> GetStatusAsync(int orderId)
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<PaymentStatus>(
            HttpMethod.Get, PaymentEndpoints.Status(orderId));
    }
}
